### ISSUE TITLE
Information does not apply to Azure Provider

### ISSUE THREAD

**Issue by GloverChris** [pulumi/docs#5604]
Hello Team 😊👋
This was a bit confusing for us in the community. This example does not apply to the Azure provider as you have access to all regions regardless of the default region you define. Region is explicitly defined in each resources see 1️⃣ Example of VNet. See 2️⃣ for some context from community slack on the confusion. Took a bit of looking around but we were able to figure it out 🔍. Any who hope some clarification can get added 🐱‍🏍.

File: [docs/intro/concepts/resources.md](https://www.pulumi.com/docs/intro/concepts/resources/#explicit-provider-configuration)

1️⃣ https://www.pulumi.com/docs/reference/pkg/azure/network/virtualnetwork/#example-usage
2️⃣ https://pulumi-community.slack.com/archives/C84L4E3N1/p1617124285065700

**Comment by mikhailshilkov** [comment]
Hi @GloverChris Thank you for reporting this.

To be clear, the idea of Explicit Providers still applies to Azure, it's just that the Azure providers don't have anything like `region` configuration at the moment.

Do I understand correctly that you are suggesting picking up another configuration property for this example? It might be tricky to find one that would apply consistently across many providers... Or do you have other suggestions?

**Comment by GloverChris** [comment]
What I'm saying is that the region Args don't exist in the azure provider however you specify region of resources when you declare them. If you look at Virtual Network under Azure Native, you will see that you declare which region you want the Vnet there rather than having declaring the region separately as part of the provider class. This means that the azure provider has access to all regions regardless of the default region which technically makes default region declaration moot for azure projects. I hope I explained this well. 😅






> Hi @GloverChris Thank you for reporting this.
> 
> 
> 
> To be clear, the idea of Explicit Providers still applies to Azure, it's just that the Azure providers don't have anything like `region` configuration at the moment.
> 
> 
> 
> Do I understand correctly that you are suggesting picking up another configuration property for this example? It might be tricky to find one that would apply consistently across many providers... Or do you have other suggestions?

**Comment by mikhailshilkov** [comment]
Yes, I understand all of that. Moreover, you can't set a region as a property of the Azure provider.

I guess I'm failing to link this to a potential improvement in the docs. FWIW, the doc clearly uses the AWS provider and doesn't claim this example generalizes to other providers. How would you change the doc to make this more clear?

**Comment by GloverChris** [comment]
Well it's in the general information section of how to use pulumi which is supposed to be agnostic to any specific language. This instead would then probably be better living in the AWS provider docs.



> Yes, I understand all of that. Moreover, you can't set a region as a property of an explicit Azure provider.
> 
> 
> 
> I guess I'm failing to link this to a potential improvement in the docs. FWIW, the doc clearly uses the AWS provider and doesn't claim this example generalizes to other providers. How would you change the doc to make this more clear?
